workspace/graph-sequencer/test/index.ts

```diff
@@ -0,0 +1,366 @@
+    [1, [3]],
+    [2, []],
+    [3, []],
+  ]), [0, 1, 2, 3])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [[2, 3], [0, 1]],
+      cycles: [],
+    }
+  )
+})
+
+test('graph with resolved cycle', () => {
+  expect(graphSequencer(new Map([
+    [0, [1]],
+    [1, [2]],
+    [2, [3]],
+    [3, [0]],
+  ]), [0, 1, 2, 3])).toStrictEqual(
+    {
+      safe: false,
+      chunks: [[0, 1, 2, 3]],
+      cycles: [[0, 1, 2, 3]],
+    }
+  )
+})
+
+test('graph with resolved cycle sub graph', () => {
+  expect(graphSequencer(new Map([
+    [0, [1]],
+    [1, [2]],
+    [2, [3]],
+    [3, [0]],
+  ]), [0, 1, 2])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [[2], [1], [0]],
+      cycles: [],
+    }
+  )
+})
+
+test('graph with resolved cycle with multiple unblocked deps', () => {
+  expect(graphSequencer(new Map([
+    [0, [3]],
+    [1, [3]],
+    [2, [3]],
+    [3, [0]],
+  ]), [0, 1, 2, 3])).toStrictEqual(
+    {
+      safe: false,
+      chunks: [
+        [0, 3],
+        [1, 2],
+      ],
+      cycles: [[0, 3]],
+    }
+  )
+})
+
+test('graph with resolved cycle with multiple unblocked deps sub graph', () => {
+  expect(graphSequencer(new Map([
+    [0, [3]],
+    [1, [3]],
+    [2, [3]],
+    [3, [0]],
+  ]), [0, 1, 2])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [
+        [0, 1, 2],
+      ],
+      cycles: [],
+    }
+  )
+})
+
+test('graph with two cycles', () => {
+  expect(graphSequencer(new Map([
+    [0, [1]],
+    [1, [0]],
+    [2, [3]],
+    [3, [2]],
+  ]), [0, 1, 2, 3])).toStrictEqual(
+    {
+      safe: false,
+      chunks: [[0, 1, 2, 3]],
+      cycles: [
+        [0, 1],
+        [2, 3],
+      ],
+    }
+  )
+})
+
+test('graph with multiple cycles', () => {
+  expect(graphSequencer(new Map([
+    [0, [2]],
+    [1, [0, 3]],
+    [2, [1]],
+    [3, [2, 4]],
+    [4, []],
+  ]), [0, 1, 2, 3, 4])).toStrictEqual(
+    {
+      safe: false,
+      chunks: [[4], [0, 2, 1], [3]],
+      cycles: [[0, 2, 1]],
+    }
+  )
+})
+
+test('graph with multiple cycles', () => {
+  expect(graphSequencer(new Map([
+    [0, [1]],
+    [1, [3]],
+    [2, []],
+    [3, [1, 2]],
+  ]), [0, 1, 2, 3])).toStrictEqual(
+    {
+      safe: false,
+      chunks: [[2], [1, 3], [0]],
+      cycles: [[1, 3]],
+    }
+  )
+})
+
+test('graph with full conn', () => {
+  expect(graphSequencer(new Map([
+    [0, [1, 2, 3]],
+    [1, [0, 2, 3]],
+    [2, [0, 1, 3]],
+    [3, [0, 1, 2]],
+    [4, [1]],
+  ]), [0, 1, 2, 3, 4])).toStrictEqual(
+    {
+      safe: false,
+      chunks: [[0, 1, 2, 3], [4]],
+      cycles: [
+        [0, 1],
+        [2, 3],
+      ],
+    }
+  )
+})
+
+test('graph with full conn sub graph', () => {
+  expect(graphSequencer(new Map([
+    [0, [1, 2, 3]],
+    [1, [0, 2, 3]],
+    [2, [0, 1, 3]],
+    [3, [0, 1, 2]],
+    [4, [1]],
+  ]), [1, 4])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [[1], [4]],
+      cycles: [],
+    }
+  )
+})
+
+test('graph with full conn sub graph', () => {
```

Comment:
don't use shortenings. What is conn? "a fully connected subgraph"? Also, write subgraph together.